presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -7012,4 +7012,10 @@ public void testCaseToMapOptimization()
                 "sum(case when  orderstatus||'-'|| comment='F' THEN  totalprice END)" +
                 "FROM orders group by 1,2").getOnlyValue().toString().indexOf("element_at"), -1);
     }
+
+    @Test
+    public void testLikePrefix()
+    {
+        assertQuery("select x like 'abc%' from (values 'abc', 'def', 'bcd') T(x)");
```

Comment:
Can you add more test cases like:

select x like 'abc%' from (values 'abc', 'abcd', 'abd') T(x)
select x like 'ab_%' from (values 'abc', 'abcd', 'abd') T(x)